src/Reflection/Adapter/ReflectionIntersectionType.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Reflection\Adapter;
+
+use ReflectionUnionType as CoreReflectionUnionType;
+use Roave\BetterReflection\Reflection\ReflectionIntersectionType as BetterReflectionIntersectionType;
+use Roave\BetterReflection\Reflection\ReflectionNamedType as BetterReflectionNamedType;
+use Roave\BetterReflection\Reflection\ReflectionUnionType as BetterReflectionUnionType;
+
+use function array_filter;
+use function array_map;
+
+class ReflectionIntersectionType extends CoreReflectionUnionType
```

Comment:
@WyriHaximus The `extends` looks like a bug.

Reply:
@kukulich It is, will push a fixed commit later tonight

Reply:
@WyriHaximus I've just fixed it. It was a little complicated to pursuade PHPStan/Psalm/PHPunit to be happy, see https://github.com/Roave/BetterReflection/pull/767/commits/e7c4e42a4cb690fc17537a2ca3da3fa4c1413d37

It may be useful for you when solving `ReflectionEnum` etc.